backend-comparison/src/burnbenchapp/base.rs

```diff
@@ -81,45 +99,94 @@ pub(crate) enum BenchmarkValues {
     Unary,
 }
 
-pub fn run() {
+pub fn execute() {
     let args = Args::parse();
-
     match args.command {
-        Commands::List => {
-            println!("Available Backends:");
-            for backend in BackendValues::iter() {
-                println!("- {}", backend);
-            }
-
-            println!("\nAvailable Benchmarks:");
-            for bench in BenchmarkValues::iter() {
-                println!("- {}", bench);
-            }
+        Commands::Auth => command_auth(),
+        Commands::List => command_list(),
+        Commands::Run(run_args) => command_run(run_args),
+    }
+}
+
+/// Create an access token from GitHub Burnbench application and store it
+/// to be used with the user benchmark backend.
+fn command_auth() {
+    let mut flow = match DeviceFlow::start(CLIENT_ID, None) {
+        Ok(flow) => flow,
+        Err(e) => {
+            eprintln!("Error authenticating: {}", e);
+            return;
         }
-        Commands::Run(run_args) => {
-            if run_args.backends.is_empty() || run_args.benches.is_empty() {
-                println!("No backends or benchmarks specified. Please select at least one backend and one benchmark.");
-                return;
-            }
-
-            let total_combinations = run_args.backends.len() * run_args.benches.len();
-            println!(
-                "Executing the following benchmark and backend combinations (Total: {}):",
-                total_combinations
-            );
-            for backend in &run_args.backends {
-                for bench in &run_args.benches {
-                    println!("- Benchmark: {}, Backend: {}", bench, backend);
-                }
-            }
-
-            let mut app = App::new();
-            app.init();
-            println!("Running benchmarks...");
-            app.run(&run_args.benches, &run_args.backends);
-            app.cleanup();
-            println!("Cleanup completed. Benchmark run(s) finished.");
+    };
+    println!("🌐 Please visit for following URL in your browser (CTRL+click if your terminal supports it):");
+    println!("\n    {}\n", flow.verification_uri.clone().unwrap());
+    let user_code = flow.user_code.clone().unwrap();
+    println!("👉 And enter code: {}", &user_code);
+    if let Ok(mut clipboard) = Clipboard::new() {
+        if clipboard.set_text(user_code).is_ok() {
+            println!("📋 Code has been successfully copied to clipboard.")
+        };
+    };
+    thread::sleep(FIVE_SECONDS);
+    match flow.poll(20) {
+        Ok(creds) => {
+            save_token(&creds.token);
         }
+        Err(e) => eprint!("Authentication error: {}", e),
+    };
```

Comment:
Can't this be done in a loop with a fixed amount of retry and an exponential sleep decay?

Reply:
No it is expected to be a retry at a regular interval. I added a link to the documentation in your previous comment for the FIVE_SECONDS.